lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -2743,11 +2743,11 @@ if (!sourceList)
 
         hr = this->GetDebugContext()->RundownSourcesAndReparse(shouldPerformSourceRundown, /*shouldReparseFunctions*/ true);
 
-        // Debugger attach/detach failure is catastrophic, take down the process
-        DEBUGGER_ATTACHDETACH_FATAL_ERROR_IF_FAILED(hr);
-
         if (!this->IsClosed())
```

Comment:
Everything is inside the !closed check. I will change the check to return hr if closed and move rest of code out of if.